modules/apps/29-fee/keeper/escrow.go

```diff
@@ -26,18 +28,31 @@ func (k Keeper) EscrowPacketFee(ctx sdk.Context, identifiedFee types.IdentifiedP
 		return sdkerrors.Wrapf(types.ErrRefundAccNotFound, "account with address: %s not found", refundAcc)
 	}
 
-	coins := identifiedFee.Fee.RecvFee
-	coins = coins.Add(identifiedFee.Fee.AckFee...)
-	coins = coins.Add(identifiedFee.Fee.TimeoutFee...)
+	if k.channelKeeper.HasPacketAcknowledgement(ctx, identifiedFee.PacketId.PortId, identifiedFee.PacketId.ChannelId, identifiedFee.PacketId.Sequence) {
+		return sdkerrors.Wrapf(channeltypes.ErrAcknowledgementExists, "failed to escrow fee for packet sequence %d", identifiedFee.PacketId.Sequence)
+	}
```

Comment:
I'm confused? Packet acknowledgements only get written on the receiving chain

Reply:
I think this needs to check that a packet commitment exists (otherwise the packet hasn't been sent, got acknowledged, or got timed out)